### PR TITLE
shared/net/bind: rename ListenerConfig and ListenUpgraderConfig

### PR DESCRIPTION
**Pull request by amery**
FooListenerConfig becomes FooListener
ListenUpgraderConfig becomes UpgraderListenerConfig
ListenUpgrader becomes Upgrader

### CODE REVIEW

File: shared/net/bind/types.go [hunk 1]
```diff
@@ -2,45 +2,45 @@ package bind
...
 
-// TCPListenerConfig provides a context-aware alternative to net.ListenTCP
-type TCPListenerConfig interface {
+// TCPListener provides a context-aware alternative to net.ListenTCP
+type TCPListener interface {
 	ListenTCP(network string, laddr *net.TCPAddr) (*net.TCPListener, error)
 }
 
-// AllTCPListenerConfig is equivalent to TCPListenerConfig but takes an
+// AllTCPListener is equivalent to TCPListenerConfig but takes an
```
Comment (karasz) [inline shared/net/bind/types.go:24]: AllTCPListener canot be equivalent to TCPLIstenerConfig, only with TCPLIstener

File: shared/net/bind/types.go [hunk 2]
```diff
@@ -2,45 +2,45 @@ package bind
...
 
-// UDPListenerConfig provides a context-aware alternative to net.ListenUDP
-type UDPListenerConfig interface {
+// UDPListener provides a context-aware alternative to net.ListenUDP
+type UDPListener interface {
 	ListenUDP(network string, laddr *net.UDPAddr) (*net.UDPConn, error)
 }
 
-// AllUDPListenerConfig is equivalent to UDPListenerConfig but takes an
+// AllUDPListener is equivalent to UDPListenerConfig but takes an
```
Comment (karasz) [inline shared/net/bind/types.go:35]: Se above, you need to correct the comment to be consistent

File: shared/net/bind/upgrader.go [hunk 3]
```diff
@@ -5,53 +5,53 @@ import (
...
 		callback func(network, addr string) (net.Listener, error)) (net.Listener, error)
 	ListenPacketWithCallback(network, addr string,
 		callback func(netowkr, addr string) (net.PacketConn, error)) (net.PacketConn, error)
 }
 
-// ListenUpgraderConfig represents an object equivalent to a ListenConfig but
-// using a ListenUpgrader in the middle
-type ListenUpgraderConfig struct {
+// UpgraderListenConfig represents an object equivalent to a ListenConfig but
+// using a UpgraderListen in the middle
```
Comment (karasz) [inline shared/net/bind/upgrader.go:26]: So its UpgraderListen or just Upgrader?

**Review by karasz — CHANGES_REQUESTED** [review]

**Comment by amery** [comment]
amended @karasz

File: shared/net/bind/types.go (hunk 1 above)
Comment (amery) [reply shared/net/bind/types.go]: amended

File: shared/net/bind/types.go (hunk 2 above)
Comment (amery) [reply shared/net/bind/types.go]: amended

**Review by karasz — APPROVED** [review]
now LGTM

**Pull request merged**